src/name_core.jl

```diff
@@ -162,26 +162,36 @@ This is the type of name combination used for binary array operations.
 Where the dimensions of both arrays must be the same.
 """
 function unify_names(names_a, names_b)
-    # 0-Allocations if inputs are the same
-    # 0-Allocation, if has a `:_` see  `@btime (()->unify_names((:a, :b), (:a, :_)))()`
-
+    # @btime (()->unify_names((:a, :b), (:a, :_)))()
     names_a === names_b && return names_a
+    length(names_a) == length(names_b) || incompatible_dimension_error(names_a, names_b)
```

Comment:
Should  this line be in `try_unify_names`  ?
(returning `nothing` rather than erroring)

Reply:
Good point, I've now moved these things around as suggested.